internal/server/server.go

```diff
@@ -0,0 +1,429 @@
+package server
```

Comment:
this is a very huge file. We should split this into multiple files. I also see duplicated code, we should refactor the code such us the validation in the server and the cli is re-used
for the folder organization we can have a folder called server:
```
server/
  server.go or router.go
  agent/
  document/
```